docs/api-spec-swagger.yaml

```diff
@@ -0,0 +1,3370 @@
+      summary: Delete a resource
+      security:
+        - jwt: []
+      parameters:
+        - name: id
+          in: path
+          description: ID of resource to delete
+          required: true
+          schema:
+            type: string
+            example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+      responses:
+        '204':
+          description: Resource deleted successfully
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: "Resource deleted."
+        '401':
+          description: Unauthorized - User is not authenticated
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: "Authentication required."            
+                    
+        '403':
+          description: Forbidden - User does not have the necessary permissions
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: "You do not have permission to access this resource."           
+        '404':
+          description: Resource not found
+          content:
+            application/json:
+              schema:
+                type: string
+                properties:
+                  message:
+                   type: string
+                   example: Resource not found.
+    
+  /resources/{id}/views:
+    get:
+      tags:
+        - resource
+      summary: Get view count for a resource
+      parameters:
+        - name: id
+          in: path
+          description: ID of resource
+          required: true
+          schema:
+            type: string
+            example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+      responses:
+        '200':
+          description: success
+          content:
+            application/json:
+              schema:
+                  type: object
+                  properties:
+                    data:
+                      type: object
+                      properties:
+                         resource_id:
+                           type: string
+                           example: f21e77c9-8fd6-4d58-ae91-2560b53c2389
+                         view_count:
+                           type: integer
+                           example: 1000
+        '404':
+          description: resource not found
+          content:
+           application/json:
+              schema:
+                  type: object
+                  properties:
+                    message:
+                      type: string
+                      example: Resource not found.
+                      
+  /resources/{resourceId}/save:

```

Comment:
I would move the userid to the url part. /resources/{resourceId}/save/{userId} is more concise.